pydra/engine/tests/utils.py

```diff
@@ -165,41 +205,43 @@ def identity_2flds(
 
 
 @mark.task
-def ten(x):
+def ten(x) -> int:
     return 10
 
 
 @mark.task
-def add2_wait(x):
+def add2_wait(x: int) -> int:
     time.sleep(2)
     return x + 2
 
 
 @mark.task
-def list_output(x):
+def list_output(x: int) -> ty.List[int]:
     return [x, 2 * x, 3 * x]
 
 
 @mark.task
-def list_sum(x):
+def list_sum(x: ty.Sequence[ty.Union[int, float]]) -> ty.Union[int, float]:
     return sum(x)
```

Comment:
Have we tested on any generics, like:

```suggestion
NumT = ty.TypeVar('NumT', bound=float)

def list_sum(x: ty.Sequence[NumT]) -> NumT:
    return sum(x)
```

These sorts of functions definitely come up, so it might be nice to make sure we can handle them, though that can be a future PR.

Reply:
I have created this issue detailing what would be required to support this, https://github.com/nipype/pydra/issues/672. Shouldn't be too much work, maybe a day or two all going well